src/mudlet-lua/lua/Other.lua

```diff
@@ -846,3 +846,69 @@ do
     end
   end
 end
+
+local timeframetable = {}
+
+function timeframe(vname, true_time, nil_time, ...)
+  local format = string.format
+
+  assert(type(true_time) == "number" or type(true_time) == "table", format("timeframe: bad argument #2 type (true time as a number or table expected, got %s!", type(true_time)))
+  assert(type(nil_time) == "nil" or type(nil_time) == "number" or type(nil_time) == "table", format("timeframe: bad argument #3 type (nil time as a number or table expected, got %s!", type(nil_time)))
+
+  -- aggregate timerlist data
+  local timerlist = {
+    {0, nil},
+    type(true_time) == "number" and {true_time, true} or type(true_time) == "table" and true_time,
+    type(nil_time) == "number" and {nil_time, nil} or type(nil_time) == "table" and nil_time,
+    ...
+  }
+
+  -- reinitialise timeframe for vname
+  killtimeframe(vname)
+  timeframetable[vname] = {}
+
+  local vtype = type(vname)
+
+  -- loop through timerlist and create tempTimers
+  local maxtime = 0
+  for step, data in ipairs(timerlist) do
+    assert(type(data) == "table", format("timeframe: bad argument #4 type (timerlist data as a table expected, got %s!", type(data)))
+    local time, value = data[1], data[2]
+
+    maxtime = (time > maxtime) and time or maxtime
+
+    local fun
+    if vtype == "function" then
+      fun = function()
+        local s,m = pcall(vname, value)
+        if not s then error(m) end
+      end
+    else
+      assert(type(value) == "string" or type(value) == "number" or type(value) == "boolean" or type(value) == "nil", format("timeframe: bad argument #4 type (timerlist data argument #2 expects a string, number or boolean value; got %s!", type(value)))
+      fun = assert(loadstring(format("%s = %s", vname, type(value) == "string" and ("'" .. value .. "'") or tostring(value))))
+    end
+
+    if time <= 0 then
```

Comment:
the type of `time` gets never checked either. And the error message is ugly: `<../src/mudlet-lua/lua/Other.lua:878: attempt to compare number with string>`